aiida_mlip/parsers/sp_parser.py

```diff
@@ -93,16 +74,19 @@ def parse(self, **kwargs) -> int:
         int
             An exit code.
         """
-        output_filename = self.node.get_option("output_filename")
-        xyzoutput = (self.node.inputs.xyz_output_name).value
-        logoutput = (self.node.inputs.log_filename).value
+
+        exit_code = super().parse(**kwargs)
+
+        if exit_code != ExitCode(0):
+            return exit_code
+
+        xyzoutput = (self.node.inputs.out).value
```

Comment:
I can't suggest changes to all the instances, but `xyz_output` would be nicer, as it's quite hard to read as is.